lib/sqlalchemy/orm/session.py

```diff
@@ -1639,6 +1646,8 @@ def __init__(
 
           .. versionadded:: 2.0.0rc1
 
+        :param close_final: Defaults to ``False``. Determines if the the session
```

Comment:
**Federico Caselli** (CaselIT) wrote:

well my initial suggestion was single_use. https://github.com/sqlalchemy/sqlalchemy/issues/7787#issuecomment-1057974309

I don't think close_final is a great name, but I don't really have other suggestions. Maybe `single_close`? but it's not really any better, so as you prefer mike

> a new method .empty() would do the old "close()" behavior (this is mentioned in the issue, likely should be added here).

empty is even worse as name at least for me. I really don't associate it with what it would do. It's probably me. Alternative that came to mind are `reset()` / `reinit()` / `discard()` (taken from the engine, since it does kind of a similar thing)

also I'm not 100% convinced about making it the new default in the future.
Meaning I can see it being default when if session is used as context manager
```py
with Session(...) as s:
  ...
```
but it seems strange to have that behaviour when the session is not used as context manager, as in
```py
s = Session(...)
...
s.close()
...
```

In any case I think at most we could start warning on 2.1, but switching the default feels like a change to be kept for at least a few releases

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4824